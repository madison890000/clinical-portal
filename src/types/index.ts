export type IPerson = {
    title?: string;
    firstName: string;
    preferredName?: string;
    middleName?: string;
    familyName: string;
    suffix?: string;
}

export type IClinician = {
    username: string;
    role: string;
} & IPerson;

export type IPatient = {
    age: number;
    sex: 'Male' | 'Female' | 'Unknown' | 'Indeterminate';
} & IPerson;

export type PatientSummary = {
    id: string;
    name: string;
}

export type IPatientList = {
    patients: PatientSummary[];
}

export type IBaseErrorResponse = {
    httpStatusCode: number;
    errorMessage: string;
}

export enum LoginStatus {
    NotLogin,
    Login,
    TokenExpired
}
