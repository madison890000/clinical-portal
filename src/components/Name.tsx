import { IPerson } from '../types';
import styles from './Name.module.scss';

export const Name = ({ title, firstName, preferredName, middleName, familyName, suffix }: IPerson) => {
    let prefixName = preferredName ? `(${preferredName})` : firstName;
    return (
        <div className={styles.container}>
            <div className={styles.text}>{title}</div>
            <div className={styles.text}>{prefixName}</div>
            <div className={styles.text}>{middleName}</div>
            <div className={styles.text}>{familyName}</div>
            <div className={styles.text}>{suffix}</div>
        </div>
    );
};

export default Name;
