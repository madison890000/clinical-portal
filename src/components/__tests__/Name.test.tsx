import React from 'react';
import Name from '../Name';
import renderer from 'react-test-renderer';
import { IPerson } from '../../types';

test('renders Name correctly', () => {
    const demoPerson = {
        username: 'joshs',
        role: 'General Practitioner',
        title: 'Dr',
        firstName: 'Joshua',
        preferredName: 'Josh',
        familyName: 'Smith'
    } as IPerson;
    const NameDom = renderer.create(<Name {...demoPerson} />).toJSON();
    expect(NameDom).toMatchSnapshot();
});
