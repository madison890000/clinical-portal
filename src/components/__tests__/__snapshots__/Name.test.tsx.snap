// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Name correctly 1`] = `
<div
  className="container"
>
  <div
    className="text"
  >
    Dr
  </div>
  <div
    className="text"
  >
    (Josh)
  </div>
  <div
    className="text"
  />
  <div
    className="text"
  >
    Smith
  </div>
  <div
    className="text"
  />
</div>
`;
